### PR TITLE
Hotfix: ignore invalid params from ENV vars and INI config file

### PR DESCRIPTION
**Pull request by nuno-agostinho**
https://github.com/Ensembl/ensembl-vep/issues/1357 and related with https://github.com/Ensembl/ensembl-vep/pull/1358

One of the params defined via environment variables is being passed to VEP installation tests, even when it is from an invalid parameter. This PR allows to silently ignore invalid parameters passed from environment variables and INI config files. The validity of the params is based on GetOptions.

## Testing

The following command should ignore the parameter `--version` but include `--spdi` in `VEP command-line` in the header.

```
export VEP_VERSION=109.2
export VEP_SPDI=1
vep -i variants.vcf
```

### CODE REVIEW

File: t/testdata/fasta/Homo_sapiens.GRCh38.toplevel.test.fa.fai [hunk 1]
```diff
@@ -0,0 +1 @@
+21	46709983	57	60	61
```
Comment (nakib103) [inline t/testdata/fasta/Homo_sapiens.GRCh38.toplevel.test.fa.fai:1]: Hi @nuno-agostinho , Is this file redundant?

**Review by nakib103 — CHANGES_REQUESTED** [review]
One main issue still remains that even if someone uses a valid flag (e.g. `VEP_DIR`) the flag will be accepted (setting `config->{dir} = <some value>`) and disrupting the test and potentially a VEP run.

One solution I can think of is to -
- read ENV variable only when running from docker.
- have a flag to disable reading from ENV variable even in docker (to prevent potential unwanted reading of variable like `VEP_DIR`).

File: t/testdata/fasta/Homo_sapiens.GRCh38.toplevel.test.fa.fai (hunk 1 above)
Comment (nuno-agostinho) [reply t/testdata/fasta/Homo_sapiens.GRCh38.toplevel.test.fa.fai]: Removed the file, I don't think we need this, was just for testing.

**Comment by nuno-agostinho** [comment]
Hi @nakib103, if they use a valid environment flag, then I think that VEP should accept it in all situations. We fixed `VEP_VERSION` because it was using a deprecated argument in `INSTALL.pl` and a non-existing argument in `vep`.

> One main issue still remains that even if someone uses a valid flag (e.g. VEP_DIR) the flag will be accepted (setting config->{dir} = <some value>) and disrupting the test and potentially a VEP run.

The test will be disrupted if any extra flag is given, whether it comes from environment variables, config INI file (that is automatically read from the user folder) or the command line. The only way to fix that is by making the test more flexible and to accept passing extra arguments, but I don't think this is a priority.

> read ENV variable only when running from docker.

There is no way to confirm we are in a Docker image (and we also want this to work in Singularity, etc.). Moreover, we want advanced users to have access to this option if they want to use them in any environment.

> have a flag to disable reading from ENV variable even in docker (to prevent potential unwanted reading of variable like VEP_DIR).

To avoid unwanted reading of variables, users should simply rename their variables so they don't start with `VEP_`. It is usual for command-line programs to accept environment variables that are prefixed with the name of the program.

**Review by nakib103 — APPROVED** [review]
After discussion, we will update our doc in e110 to have a not on usage of ENV variable with VEP_ prefix. 
LOTM !~

**Pull request merged**